packages/native/types/base.d.ts

```diff
@@ -0,0 +1,206 @@
+// Definitions by: Pat Sissons <https://github.com/patsissons>
+// TypeScript Version: 3.4
+
+import { ComponentPropsWithoutRef, ComponentType } from 'react'
+import * as RN from 'react-native'
+
+type ReactNative = typeof RN
+
+export type ReactNativeStyle = ReturnType<ReactNative['StyleSheet']['flatten']>
+export type ObjectInterpolation = RN.ViewStyle | RN.TextStyle | RN.ImageStyle
```

Comment:
this type union should be fine instead of the polymorphic version that was being used previously.

Reply:
Seems like it's hard for us to create strict typings for those if we want to have types similar to those for the web version. From what I understand this will indeed work somewhat OK, but will allow incorrect object styles to be set on a particular component (in example it will allow text styles to be used on a view component). To keep things simple, I would say it's OK. WDYT?

Reply:
I'll take another pass to see if I can improve it, but I think that may not be avoidable